src/node_process_object.cc

```diff
@@ -116,6 +115,35 @@ MaybeLocal<Object> CreateProcessObject(Realm* realm) {
   }
   NODE_VERSIONS_KEYS(V)
 #undef V
+  // todo: sort the keys except node
+  Local<v8::Array> keys = versions->GetOwnPropertyNames(context).ToLocalChecked();
+  uint32_t keys_length = keys->Length();
+  std::vector<std::string> version_keys;
+  for (uint32_t i = 0; i < keys_length; i++) {
+    Local<Value> key = keys->Get(context, i).ToLocalChecked();
+    if (!key->IsString()) continue;
+    Utf8Value key_str(isolate, key.As<v8::String>());
+    version_keys.push_back(std::string(*key_str));
+  }
+  std::sort(version_keys.begin(), version_keys.end(),
+            [](const std::string& a, const std::string& b) {
+              if (a == "node") {
+                return true;
+              }
+              return a < b;
+            });
```

Comment:
It probably makes more sense to modify the macro expansion above than to generate a JS object in C++ only to consume, re-create and discard it in C++

Reply:
I'm not sure how to maintain a sorted object when you `object->Set` increasingly

Reply:
Right, the idea would be to sort the keys *before* creating a JS object :)

Reply:
make sense, thanks for sugesstion